R/detect_mistakes.R

```diff
@@ -203,9 +204,30 @@ detect_mistakes <- function(user,
       )
     }
 
+    
+    if (length(well_matched > 0)){
+      matched_user_names <- sort(rlang::names2(well_matched))
+      
+    if ( !allow_partial_matching ){
+      
+      matches <- which(sapply(lapply(remaining_solution_names,startsWith, matched_user_names),sum)==1)
+      matched_solution_name <- sort(remaining_solution_names[matches])
+      
+      return( 
+        pmatches_argument_name(
+          this_call = user,
+          this = as.character(user[matched_user_names]),
```

Comment:
Does `this` allow for vectors?

Same with `this_name` and `correct_name`

Reply:
Could only report the first bad name instead of all of the bad names

Reply:
all bad names are listed

![image](https://user-images.githubusercontent.com/10470699/97505305-fa3cf180-1978-11eb-9313-5508c2a28153.png)
this seems to me to be the best solution

Reply:
@garrettgman Thoughts on returning **first** partial match offender or **all** partial match offenders?

Reply:
I'm fine with either, but I think we should change `parameter(s)` in the message to `argument(s)` to accord with common usage.

Reply:
done here ; https://github.com/rstudio-education/gradethis/pull/161/commits/7d9f9fd85ed61ce0c611ab6ba7a856535d2eb26d